modules/core/src/shaderlib/project/viewport-uniforms.js

```diff
@@ -172,26 +172,23 @@ export function getUniformsFromViewport({
 } = {}) {
   assert(viewport);
 
-  return Object.assign(
-    {
-      project_uModelMatrix: modelMatrix || IDENTITY_MATRIX
-    },
-    getMemoizedViewportUniforms({
-      viewport,
-      devicePixelRatio,
-      coordinateSystem,
-      coordinateOrigin,
-      wrapLongitude
-    })
-  );
+  return getMemoizedViewportUniforms({
```

Comment:
Adding more parameters to the memoized function makes it more likely to recalculate. 

Maybe we can mutate the cached result by assigning `modelMatrix` to it — it’s not “safe” but we are only using the object once and immediately.